docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,958 @@
+| `ControlPlane.AvailableReplicas` (new)   | `ControlPlane.AvailableReplicas`                         | `ControlPlane.AvailableReplicas`           |
+| `Workers` (new)                          | `Workers`                                                | `Workers`                                  |
+| `Workers.DesiredReplicas` (new)          | `Workers.DesiredReplicas`                                | `Workers.DesiredReplicas`                  |
+| `Workers.Replicas` (new)                 | `Workers.Replicas`                                       | `Workers.Replicas`                         |
+| `Workers.ReadyReplicas` (new)            | `Workers.ReadyReplicas`                                  | `Workers.ReadyReplicas`                    |
+| `Workers.UpToDateReplicas` (new)         | `Workers.UpToDateReplicas`                               | `Workers.UpToDateReplicas`                 |
+| `Workers.AvailableReplicas` (new)        | `Workers.AvailableReplicas`                              | `Workers.AvailableReplicas`                |
+| other fields...                          | other fields...                                          | other fields...                            |
+
+notes:
+- The `BackCompatibilty` struct is going to exist in v1beta2 types only until v1beta1 removal (9 months or 3 minor releases after v1beta2 is released/v1beta1 is deprecated, whichever is longer).
+  Fields in this struct are used for supporting down conversions, thus providing users relying on v1beta1 APIs additional buffer time to pick up the new changes.
+
+##### Cluster (New)Conditions
+
+| Condition                 | Note                                                                                                                                                                                                                                                                                                                  |
+|---------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `Available`               | True if Cluster `RemoteConnectionProbe` is true, if Cluster's control plane `Available` condition is true, if all MachineDeployment and MachinePool's `Available` condition are true; if conditions are defined in `spec.availabilityGates`, those conditions should be true as well for the Cluster to be available. |
+| `ControlPlaneInitialized` | True when the Cluster's control plane is functional enough to accept requests. This information is usually used as a signal for starting all the provisioning operations that depends on a functional API server, but do not require a full HA control plane to exists.                                               |
```

Comment:
I'm not clear on the relationship between this and `ClusterInitializationStatus.ControlPlaneInitialized`. When should I look at one vs. the other?

When would I look at this instead of `ControlPlaneAvailable`?

Reply:
status.Initialized.ControlPlaneInitialized and status.conditions[ControlPlaneInitialized] are the same signal, but conditions offer an overall status on the control plane which instead do not exist under status.Initialized (you can look at both, but IMO using conditions should be preferred)

More context:

status.Initialized.ControlPlaneInitialized  comes from the ClusterAPI contract, and we are preserving this field because proposal is not going to change the fact that Cluster API contract uses flags to orchestrate initial provisioning (see latest paragraphs I have added to the doc for more context, also divide and conquer 😉)

status.conditions[ControlPlaneInitialized] has been added because it can be used in combination with `kubectl wait` when writing simple scripts to automate processes around cluster provisioning

Reply:
Resolving, hopefully it is clear now